File: backend/HskLearn.Api/Models/WritingEvaluationResult.cs
namespace HskLearn.Api.Models;

public record WritingEvaluationResult(
    int OverallScore,
    string StrokeOrder,
    string Proportion,
    string Similarity,
    string RecognisedMeaning,
    List<string> Suggestions,
    string Encouragement
);

public record CompositionEvaluationResult(
    int OverallScore,
    string Transcription,
    string Grammar,
    string Vocabulary,
    string Structure,
    string Content,
    List<string> Corrections,
    List<string> Suggestions,
    string Encouragement
);

public record WritingPrompt(
    string PromptChinese,
    string PromptEnglish,
    string Topic
);

File: android/app/src/main/java/com/hsklearn/app/data/model/WritingEvaluationResult.kt
package com.hsklearn.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class WritingEvaluationResult(
    val overallScore: Int,
    val strokeOrder: String,
    val proportion: String,
    val similarity: String,
    val recognisedMeaning: String = "",
    val suggestions: List<String>,
    val encouragement: String,
)

@Serializable
data class CompositionEvaluationResult(
    val overallScore: Int,
    val transcription: String = "",
    val grammar: String,
    val vocabulary: String,
    val structure: String,
    val content: String,
    val corrections: List<String>,
    val suggestions: List<String>,
    val encouragement: String,
)

@Serializable
data class WritingPrompt(
    val promptChinese: String,
    val promptEnglish: String,
    val topic: String,
)
